spring/src/main/java/org/axonframework/spring/authorization/AuthorizationMessageDispatchInterceptor.java

```diff
@@ -0,0 +1,41 @@
+package org.axonframework.spring.authorization;
+
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.messaging.MessageDispatchInterceptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+/**
+ * Message dispatch interceptor that adds the {$code username} and {$code authorities} from the authorized principle
+ *
+ * @author Roald Bankras
+ */
+public class AuthorizationMessageDispatchInterceptor implements MessageDispatchInterceptor<CommandMessage<?>> {
+
+    private static final Logger log = LoggerFactory.getLogger(AuthorizationMessageDispatchInterceptor.class);
+
+    @Nonnull
+    @Override
+    public CommandMessage<?> handle(@Nonnull CommandMessage<?> message) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
Is there a chance any of this might throw a `NullPointerException`?

Reply:
getContext() is guaranteed never null. getAuthentication might return null if no authenticatio information is available